File: PiDEV-JAVA-Integrated-Final-v8/src/main/java/models/SuiviTherapeutique.java
package models;

import java.util.Date;

public class SuiviTherapeutique {
    private int id;
    private int patientId;       // FK → users.id (role='patient')
    private String patientNom;   // joined field for display only
    private Date dateDebut;
    private Date dateFin;
    private String typeSuivi;
    private String objectifTherapeutique;
    private String statut;

    public SuiviTherapeutique() {}

    /** Full constructor used by the service when loading from DB with JOIN */
    public SuiviTherapeutique(int id, int patientId, String patientNom,
                               Date dateDebut, Date dateFin,
                               String typeSuivi, String objectifTherapeutique, String statut) {
        this.id = id;
        this.patientId = patientId;
        this.patientNom = patientNom;
        this.dateDebut = dateDebut;
        this.dateFin = dateFin;
        this.typeSuivi = typeSuivi;
        this.objectifTherapeutique = objectifTherapeutique;
        this.statut = statut;
    }

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public int getPatientId() { return patientId; }
    public void setPatientId(int patientId) { this.patientId = patientId; }

    public String getPatientNom() { return patientNom; }
    public void setPatientNom(String patientNom) { this.patientNom = patientNom; }

    public Date getDateDebut() { return dateDebut; }
    public void setDateDebut(Date dateDebut) { this.dateDebut = dateDebut; }

    public Date getDateFin() { return dateFin; }
    public void setDateFin(Date dateFin) { this.dateFin = dateFin; }

    public String getTypeSuivi() { return typeSuivi; }
    public void setTypeSuivi(String typeSuivi) { this.typeSuivi = typeSuivi; }

    public String getObjectifTherapeutique() { return objectifTherapeutique; }
    public void setObjectifTherapeutique(String objectifTherapeutique) { this.objectifTherapeutique = objectifTherapeutique; }

    public String getStatut() { return statut; }
    public void setStatut(String statut) { this.statut = statut; }

    @Override
    public String toString() { return typeSuivi + " - " + statut; }
}

File: symfony-integ/src/Entity/SuiviTherapeutique.php
<?php
namespace App\Entity;
use App\Repository\SuiviTherapeutiqueRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: SuiviTherapeutiqueRepository::class)]
#[ORM\Table(name: 'suivi_therapeutique')]
class SuiviTherapeutique
{
    #[ORM\Id] #[ORM\GeneratedValue] #[ORM\Column]
    private ?int $id = null;
    #[ORM\Column(name: 'patient_id')]                         private ?int $patientId = null;
    #[ORM\Column(name: 'date_debut', type: 'datetime')]       private ?\DateTimeInterface $dateDebut = null;
    #[ORM\Column(name: 'date_fin', type: 'datetime', nullable: true)] private ?\DateTimeInterface $dateFin = null;
    #[ORM\Column(name: 'type_suivi', length: 100)]            private ?string $typeSuivi = null;
    #[ORM\Column(name: 'objectif_therapeutique', length: 255)] private ?string $objectifTherapeutique = null;
    #[ORM\Column(length: 50)]                                  private ?string $statut = null;

    private ?string $patientNom = null;

    public function getId(): ?int { return $this->id; }
    public function getPatientId(): ?int { return $this->patientId; }
    public function setPatientId(int $v): static { $this->patientId = $v; return $this; }
    public function getDateDebut(): ?\DateTimeInterface { return $this->dateDebut; }
    public function setDateDebut(\DateTimeInterface $v): static { $this->dateDebut = $v; return $this; }
    public function getDateFin(): ?\DateTimeInterface { return $this->dateFin; }
    public function setDateFin(?\DateTimeInterface $v): static { $this->dateFin = $v; return $this; }
    public function getTypeSuivi(): ?string { return $this->typeSuivi; }
    public function setTypeSuivi(string $v): static { $this->typeSuivi = $v; return $this; }
    public function getObjectifTherapeutique(): ?string { return $this->objectifTherapeutique; }
    public function setObjectifTherapeutique(string $v): static { $this->objectifTherapeutique = $v; return $this; }
    public function getStatut(): ?string { return $this->statut; }
    public function setStatut(string $v): static { $this->statut = $v; return $this; }
    public function getPatientNom(): ?string { return $this->patientNom; }
    public function setPatientNom(?string $v): static { $this->patientNom = $v; return $this; }
}
